src/k8s/pkg/k8sd/features/cilium/cleanup.go

```diff
@@ -18,5 +19,28 @@ func CleanupNetwork(ctx context.Context, snap snap.Snap) error {
 		}
 	}
 
+	for _, cmd := range []string{"iptables", "ip6tables", "iptables-legacy", "ip6tables-legacy"} {
+		out, err := exec.Command(fmt.Sprintf("%s-save", cmd)).Output()
+		if err != nil {
+			return fmt.Errorf("failed to read iptables rules: %w", err)
+		}
+
+		lines := strings.Split(string(out), "\n")
+		for i, line := range lines {
+			for _, word := range []string{"cilium", "kube", "CILIUM", "KUBE"} {
+				if strings.Contains(line, word) {
+					lines[i] = ""
+					break
+				}
+			}
```

Comment:
Maybe we can go with the following?
(Not sure if it's okay to count words like `Kubernetes` or `Cilium` as well).
```suggestion
			for _, word := range []string{"cilium", "kube"} {
				if strings.Contains(strings.ToLower(line), word) {
					lines[i] = ""
					break
				}
			}
```